src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/ClassMemberImpliedModifierCheck.java

```diff
@@ -0,0 +1,200 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2018 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.modifier;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.ScopeUtil;
+
+/**
+ * <p>
+ * Checks for implicit modifiers on nested types in classes.
```

Comment:
in classes and enums ....

@rnveach , should be count enum as class or it is special type .... so in this case Check should be named "TypeMember.... "

Reply:
The [JLS](https://docs.oracle.com/javase/specs/jls/se10/html/) considers enums to be a kind of class.

Reply:
from your code:
> `if (ScopeUtil.isInClassBlock(ast) || ScopeUtil.isInEnumBlock(ast)) {`

https://github.com/checkstyle/checkstyle/pull/6067/files/5d14b3055a509afd86bb14071eae9a72df1d0ff6#diff-637f7830124c584d7868e0dbae22e4abR162

what about enums inside enums ... weird stuff ... https://stackoverflow.com/a/8941089/1015848 but possible.

Reply:
The main issue doesn't state that this is only for special parents. It is just saying where it is nested and it is implied static.
I would say anywhere there is a nested interface/enum should be violated on then that meat the implied definition. If user wants to restrict between class/eunm, they can use xpath suppression.

Reply:
> not enums at the top level

Nothing top level can be implied static, so that is the reason why there should be no violation.

https://docs.oracle.com/javase/specs/jls/se10/html/jls-8.html#jls-ClassModifier
>  The modifier static pertains only to member classes (§8.5.1), not to top level or local or anonymous classes.

Reply:
My test case has a nested enum inside an enum inside a class.
https://github.com/checkstyle/checkstyle/pull/6067/files/5d14b3055a509afd86bb14071eae9a72df1d0ff6#diff-85905f481e3acb7907da4851a6de9d94R28

I don't think I need any changes from the comments above?

Reply:
> If user wants to restrict between class/eunm parent, they can use xpath suppression.

good catch, we almost forgot to add test to prove xpath support.

@jodastephen , you can find examples at https://github.com/checkstyle/checkstyle/tree/master/src/it/java/org/checkstyle/suppressionxpathfilter 
Sorry that we did not raised this in advance, such tests were introduced recently ... ALL checks will follow this, but NEW Checks must follow it.
It should also be added to list https://checkstyle.org/config_filters.html#SuppressionXpathFilter
source is at https://github.com/checkstyle/checkstyle/blob/master/src/xdocs/config_filters.xml#L663

Reply:
Name of the check should stay the same. I call back my original question.

Reply:
@romani If nothing is to be done here, try marking this as `resolved` right below. I assume it will hide the conversation so it doesn't look like its still open.
If there is still something to be changed, please point back to so it is clear what it is.